crates/runtime/src/datafusion.rs

```diff
@@ -607,6 +612,34 @@ impl DataFusion {
 
         Ok(())
     }
+
+    /// Returns all tables that are in the public schema.
+    #[allow(clippy::redundant_closure_for_method_calls)]
+    #[allow(clippy::flat_map_option)]
+    pub async fn get_public_table_names(&self) -> Result<Vec<String>> {
+        let rb = self
+            .ctx
+            .sql("SELECT table_name FROM information_schema.tables WHERE table_schema = 'public'")
+            .await
+            .context(UnableToGetTablesSnafu)?
+            .collect()
+            .await
+            .context(UnableToGetTablesSnafu)?;
+
+        Ok(rb
+            .iter()
+            .flat_map(|rb| {
+                rb.column(0)
+                    .as_any()
+                    .downcast_ref::<StringArray>()
+                    .map(|ar| {
+                        ar.iter()
+                            .filter_map(|x: Option<&str>| x.map(|x| x.to_string()))
+                    })
+            })
+            .flatten()
+            .collect_vec())
+    }
```

Comment:
```suggestion
    pub async fn get_public_table_names(&self) -> Result<Vec<String>> {
        Ok(self
            .ctx
            .catalog(SPICE_DEFAULT_CATALOG)
            .context(CatalogMissingSnafu {
                catalog: SPICE_DEFAULT_CATALOG.to_string(),
            })?
            .schema(SPICE_DEFAULT_SCHEMA)
            .context(SchemaMissingSnafu {
                schema: SPICE_DEFAULT_SCHEMA.to_string(),
            })?
            .table_names())
    }
```

Datafusion already provides a way to get the list of tables in a schema.

For the above code to work, you can add this to the error enum:

```rust
    #[snafu(display("The catalog {catalog} is not registered."))]
    CatalogMissing { catalog: String },

    #[snafu(display("The schema {schema} is not registered."))]
    SchemaMissing { schema: String },
```